### PR TITLE
fix(gradle-wrapper): docker java version downgrade to Java 8 to suppo…

### LINKED ISSUE #5974: Gradle wrapper update fails for gradle-wrapper v. 2.x
**What Renovate type are you using?**

Self-hosted

**Describe the bug**
Renovate fails to update gradle wrapper that has version `2.x.`, because gradle `2.x` does not support `Java 11`.

**Did you see anything helpful in debug logs?**

`Could not determine java version from '11.0.5'.`

**To Reproduce**

1. Set up gradle wrapper `v. 2.x` and install Java 11
2. Run renovate

Expected:
`gradle-wrapper.jar,` `gradlew`, `gradlew.bat`, `gradle-wrapper.properties` files are updated.

Actual:
Only `gradle-wrapper.properties` file is updated.
Gradle logs an error: `Could not determine java version from '11.0.5'.`

**Proposed fix**

Replace `Java 11` with `Java 8`

In Dockerfile:

`openjdk-11-jre-headless` -> `openjdk-8-jre-headless`

Reference:
https://github.com/facebook/react-native/issues/22487#issuecomment-443759776

### PR DESCRIPTION
**Pull request by driver733**
Closes #5974

### CODE REVIEW

**Comment by viceice** [comment]
If we do this, we need to extend our ci to test for java8 too

**Comment by viceice** [comment]
https://github.com/renovatebot/renovate/blob/e84528086e73dfc69e835e4fbc727c3efd3adb94/.github/workflows/build.yml#L46

**Review by viceice — CHANGES_REQUESTED** [review]
we need at least to enable tests on java 8 too. this would extend ci time, because of three new jobs 😢

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
         os: [ubuntu-latest, macos-latest, windows-latest]
         node-version: [12, 10]
         python-version: [3.8]
-        java-version: [11]
+        java-version: [8, 11]
         exclude:
           - os: windows-latest
             node-version: 10
```
Comment (viceice) [inline .github/workflows/build.yml:49]:
```suggestion
            node-version: 10
          # exclude java 8 on node 10
          - java-version: 8
            node-version: 10
```

Comment (rarkins) [reply .github/workflows/build.yml:49]: I prefer not to add any more jobs

**Comment by driver733** [comment]
@viceice @rarkins I think we can simply replace `Java 11` with `Java 8` in CI build. What do you think?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -43,10 +43,13 @@ jobs:
...
+          # exclude java 8 on node 10
+          - java-version: 8
+            node-version: 10
```
Comment (viceice) [inline .github/workflows/build.yml:52]:
```suggestion
            node-version: 10
          # exclude java 11 on node 10
          - java-version: 11
            node-version: 10
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -43,10 +43,16 @@ jobs:
...
+          # exclude java 11 on node 10
+          - java-version: 11
+            node-version: 10
```
Comment (viceice) [inline .github/workflows/build.yml:55]:
```suggestion
        java-version: [8]
        exclude:
          - os: windows-latest
            node-version: 10
```

**Comment by ps-feng** [comment]
I'm not sure if it's due to this change but since updating we're seeing this error with our Kotlin projects:
```
"Command failed: ./gradlew --init-script renovate-plugin.gradle renovate\n\nFAILURE: Build failed with an exception.\n\n* What went wrong:\nExecution failed for task ':buildSrc:compileKotlin'.\n> Kotlin could not find the required JDK tools in the Java installation '/usr/lib/jvm/java-8-openjdk-amd64/jre' used by Gradle. Make sure Gradle is running on a JDK, not JRE.\n\n* Try:\nRun with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.\n\n* Get more help at https://help.gradle.org\n\nBUILD FAILED in 1s\n"
```
Do you know what could be the problem?

**Comment by viceice** [comment]
@ps-feng the `:buildSrc:compileKotlin` task should not be called. Can you create a public repro and open a new issue?

**Comment by ps-feng** [comment]
OK, please bear with me because I'll need to investigate how, as we're using a self-hosted runner within our private Github that auto-updates itself (currently on 19.212.1).

**Comment by ps-feng** [comment]
@viceice Still don't have a public repo, but I can confirm that when you use the `kotlin-dsl` plugin it'll run `:buildSrc:compileKotlin`. That's probably because in `buildSrc` you can have custom Kotlin code and it needs to compile it. We use it to have dependency versions at the moment.

To check it I copied the plugin from [here](https://github.com/renovatebot/renovate/blob/4f59b62da40a653352c839b344cbfbbf9597f98a/lib/manager/gradle/gradle-updates-report.ts) and placed it in a `renovate-plugin.gradle` file. Then ran
```bash
 $ ./gradlew --init-script renovate-plugin.gradle renovate
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :buildSrc:compileKotlin
The `kotlin-dsl` plugin applied to project ':buildSrc' enables experimental Kotlin compiler features. For more information see https://docs.gradle.org/6.3/userguide/kotlin_dsl.html#sec:kotlin-dsl_plugin

BUILD SUCCESSFUL in 21s
15 actionable tasks: 15 executed
<-------------> 0% WAITING
> IDLE

```

**Pull request merged**